src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -148,43 +149,158 @@ private static int getUserCount(Request req, Response res) {
+            }
+            return result;
+        }
+    }
+
+    /**
+     * Class to hold a user's profile and raw JSON permissions.
+     */
+    public static class Auth0UserSearchResult {
+        /**
+         * If the tenant user already has permissions related to another instance of datatools they are preserved here.
+         * This will prevent them from being overwritten when creating permissions for other datatools instances.
+         */
+        String existingPermissions = "";
+
+        /**
+         * Used to hold email, userId and expanded datatools parameters if they match the AUTH0_CLIENT_ID parameter.
+         */
+        Auth0UserProfile user;
+    }
+
+    /**
+     * Create user request.
+     */
+    private static String createUserRequest(Request req, String email, String password) {
+        HttpPost createUserRequest = new HttpPost(baseUsersUrl);
+        setHeaders(req, createUserRequest);
+        String requestBody = createUserRequestBody(email, password);
+        setRequestEntityUsingJson(createUserRequest, requestBody, req);
         return executeRequestAndGetResult(createUserRequest, req);
     }
 
+    /**
+     * Produce JSON request body for create public user with default/blank permissions.
+     */
+    private static String createUserRequestBody(String email, String password) {
+        ArrayNode emptyArrayNode = mapper.createArrayNode();
+        ObjectNode permissions = mapper.createObjectNode();
+        permissions.set("permissions", emptyArrayNode);
+        permissions.set("projects", emptyArrayNode);
+        permissions.set("subscriptions", emptyArrayNode);
+        permissions.put("client_id", AUTH0_CLIENT_ID);
+        ArrayNode dataTools = mapper.createArrayNode();
+        dataTools.add(permissions);
+        ObjectNode appMetaData = mapper.createObjectNode();
+        appMetaData.set("datatools", dataTools);
+        ObjectNode createUserRequestBody = mapper.createObjectNode();
+        createUserRequestBody.put("connection", "Username-Password-Authentication");
+        createUserRequestBody.put("email", email);
+        createUserRequestBody.put("password", password);
+        createUserRequestBody.set("app_metadata", appMetaData);
+        return createUserRequestBody.toString();
+    }
+
+    /**
+     * Update user permissions. Only used in cases where the Auth0 user already exists and permissions for the current
+     * instance of datatools has not been set previously.
+     */
+    private static String updateUserPermissions(Request req, Auth0UserSearchResult searchResult, String newPermissions) {
+        // if there are existing permissions the assumption is that the datatools info has been set, but, only if it
+        // matches the current instance of datatools (AUTH0_CLIENT_ID).
+        if(!searchResult.existingPermissions.isEmpty() && searchResult.user.getApp_metadata().getDatatoolsInfo() != null) {
```

Comment:
I'm not entirely following the conditions where this block will evaluate to true.